### ISSUE TITLE
os.path.exists crashes pyodide when using PROXYFS

### ISSUE THREAD

**Issue by omert** [pyodide/pyodide#3179]
## 🐛 Bug
When file_path is in a directory that uses PROXYFS, running os.path.exists(file_path) returns True when the file exists but crashes pyodide when it does not. 

### To Reproduce

index.html:
```
<!DOCTYPE html>
<html>
    <body>
        <script> new Worker('webworker.js'); </script>
    </body>
</html>
```
webworker.js:
```
importScripts("https://cdn.jsdelivr.net/pyodide/v0.21.3/full/pyodide.js");

async function loadPyodideAndPackages() {
  self.pyodide1 = await loadPyodide();
  self.pyodide2 = await loadPyodide();

  await self.pyodide1.FS.mkdir("/dir1");
  await self.pyodide2.FS.mkdir("/dir2");
  await self.pyodide2.FS.mount(self.pyodide2.FS.filesystems.PROXYFS, {
    root: "/dir1",
    fs: self.pyodide1.FS
  }, "/dir2");

  // this code generates the expected output True False
  await self.pyodide1.runPython(`
import os

with open('/dir1/foo.txt', 'w') as f:
    f.write('hello')
print(os.path.exists('/dir1/foo.txt'));
print(os.path.exists('/dir1/bar.txt'));
`);

  
  // this code outputs True but then crashes on the last line
  await self.pyodide2.runPython(`
import os

print(os.path.exists('/dir2/foo.txt'));
print(os.path.exists('/dir2/bar.txt'));
`);

}

let pyodideReadyPromise = loadPyodideAndPackages();
```

The error message in the console is the following:
![Screenshot from 2022-10-11 21-24-11](https://user-images.githubusercontent.com/14848/195249713-a050c529-8184-4946-adab-bb59788b2ba2.png)

### Expected behavior
The code above should have produced the output
```
True
False
True
False
```
instead of crashing.
### Environment

- Pyodide Version: v0.21.3
- Browser version: Chrome 105.0.5195.125, Firefox 105.0

**Comment by hoodmane** [comment]
Thanks for the report @omert. I believe that this is an upstream bug in Emscripten @scb100.

**Comment by hoodmane** [comment]
No. Ideally someone would make an Emscripten bug report for it but doing so requires a fair amount of effort. Someone needs to track down what posix C functions are being used by `os.path.exists` and come up with a C program that reproduces the bug.

If you are willing to dig into this, it would be very helpful if you would do it. If not someone like me or @tiran will hopefully get around to it. Relatedly, it would be great to run the Python core file system tests in each type of file system. That would probably catch a few more bugs like that.

**Comment by omert** [comment]
Got it! I will probably leave this for someone else to do.

**Comment by ryanking13** [comment]
> I believe that this is an upstream bug in Emscripten @ scb100.

I think you mentioned the wrong person?

**Comment by ryanking13** [comment]
This is resolved in recent Emscripten versions, tested by #5830. Thanks @zsaladin!
